scipy/stats/tests/test_hypotests.py

```diff
@@ -1418,249 +1418,359 @@ def test_same_input(self):
         res = cramervonmises_2samp(x[:4], x[:4])
         assert_equal((res.statistic, res.pvalue), (0.0, 1.0))
 
-
-class TestTukeyHSD:
-
-    data_same_size = ([24.5, 23.5, 26.4, 27.1, 29.9],
-                      [28.4, 34.2, 29.5, 32.2, 30.1],
-                      [26.1, 28.3, 24.3, 26.2, 27.8])
-    data_diff_size = ([24.5, 23.5, 26.28, 26.4, 27.1, 29.9, 30.1, 30.1],
-                      [28.4, 34.2, 29.5, 32.2, 30.1],
-                      [26.1, 28.3, 24.3, 26.2, 27.8])
-    extreme_size = ([24.5, 23.5, 26.4],
-                    [28.4, 34.2, 29.5, 32.2, 30.1, 28.4, 34.2, 29.5, 32.2,
-                     30.1],
-                    [26.1, 28.3, 24.3, 26.2, 27.8])
-
-    sas_same_size = """
-    Comparison LowerCL Difference UpperCL Significance
-    2 - 3	0.6908830568	4.34	7.989116943	    1
-    2 - 1	0.9508830568	4.6 	8.249116943 	1
-    3 - 2	-7.989116943	-4.34	-0.6908830568	1
-    3 - 1	-3.389116943	0.26	3.909116943	    0
-    1 - 2	-8.249116943	-4.6	-0.9508830568	1
-    1 - 3	-3.909116943	-0.26	3.389116943	    0
-    """
-
-    sas_diff_size = """
-    Comparison LowerCL Difference UpperCL Significance
-    2 - 1	0.2679292645	3.645	7.022070736	    1
-    2 - 3	0.5934764007	4.34	8.086523599	    1
-    1 - 2	-7.022070736	-3.645	-0.2679292645	1
-    1 - 3	-2.682070736	0.695	4.072070736	    0
-    3 - 2	-8.086523599	-4.34	-0.5934764007	1
-    3 - 1	-4.072070736	-0.695	2.682070736	    0
+#
+# class TestTukeyHSD:
```

Comment:
Please leave all existing tests unchanged and just add new test methods for `equal_var=True`.

Please also add input validation that `equal_var` is boolean in `_tukey_hsd_iv` and add a test for it following the example of `test_is_1d` or `test_no_inf`.